torch/csrc/distributed/c10d/CUDASymmetricMemory.cu

```diff
@@ -732,7 +814,10 @@ bool CUDASymmetricMemoryAllocator::is_rendezvous_completed(void* ptr) {
 }
 
 bool CUDASymmetricMemoryAllocator::has_multicast_support() {
-  return ::has_multicast_support();
+  // Use 0 to prevent unexpected cuda context initialization.
```

Comment:
Would `[cudaGetDevice](c10::cuda::GetDevice)` make more sense here rather than hardcoding the assumption that context creation on device 0 is fine?

Reply:
The usage of this function in Python would be similar to `torch.cuda.is_available()` which can be called before a device is set (e.g., at module level). I'm worried that cudaGetDevice / c10::cuda::GetDevice would cause extra device context to be initialized on device 0. Let me think about it more.

Reply:
@Aidyn-A has done a lot of work in ensuring that the internal `GetDevice` does not eagerly create a context vs. the CUDA API call, pinging him here to see if another call should be used

Reply:
@eqy I've updated this function to receive a device_idx from the caller.